File: src/main/java/net/minestom/server/utils/async/AsyncUtils.java
package net.minestom.server.utils.async;

import org.jetbrains.annotations.ApiStatus;
import org.jetbrains.annotations.Nullable;

import java.util.concurrent.CompletableFuture;

@ApiStatus.Internal
public final class AsyncUtils {
    public static final CompletableFuture<Void> VOID_FUTURE = CompletableFuture.completedFuture(null);

    @SuppressWarnings("unchecked")
    public static <T extends @Nullable Object> CompletableFuture<T> empty() {
        return (CompletableFuture<T>) VOID_FUTURE;
    }
}

File: src/main/java/net/minestom/server/utils/async/AsyncUtils.kt
package net.minestom.server.utils.async

import java.util.concurrent.CompletableFuture
import java.lang.Void
import net.minestom.server.utils.async.AsyncUtils
import java.lang.Runnable
import net.minestom.server.MinecraftServer
import org.jetbrains.annotations.ApiStatus
import java.lang.Exception

@ApiStatus.Internal
object AsyncUtils {
    val VOID_FUTURE = CompletableFuture.completedFuture<Void?>(null)
    fun <T> empty(): CompletableFuture<T?> {
        return VOID_FUTURE as CompletableFuture<T?>
    }

    fun runAsync(runnable: Runnable): CompletableFuture<Void> {
        return CompletableFuture.runAsync {
            try {
                runnable.run()
            } catch (e: Exception) {
                MinecraftServer.getExceptionManager().handleException(e)
            }
        }
    }
}